haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -444,3 +459,52 @@ def _find_longest_common_ngram(
             # no common sequence found
             longest = ""
         return longest if longest.strip() else None
+
+    def _split_sentences(self, text: str) -> List[str]:
+        """
+        Tokenize text into sentences.
+        :param text: str, text to tokenize
+        :param language: str, language of text
+        :param tokenizer_model_path: Path, path to tokenizer model
+        :return: list[str], list of sentences
+        """
+        sentences = []
+
+        language_name = iso639_to_nltk.get(self.language)
+
+        # Try to load a custom model from 'tokenizer_model_path'
+        if self.tokenizer_model_folder:
+            tokenizer_model_path = Path(self.tokenizer_model_folder).absolute() / f"{self.language}.pickle"
+            try:
+                sentence_tokenizer = nltk.data.load(f"file:{str(tokenizer_model_path)}", format="pickle")
+                sentences = sentence_tokenizer.tokenize(text)
+            except LookupError:
+                logger.exception(f"PreProcessor couldn't load sentence tokenizer from {str(tokenizer_model_path)}")
+            except (UnpicklingError, ValueError) as e:
+                logger.exception(
+                    f"PreProcessor couldn't determine model format of sentence tokenizer at {str(tokenizer_model_path)}."
+                )
+            if sentences:
+                return sentences
+
+            # NLTK failed to split, fallback to the default model or to English
+            if language_name:
+                logger.error(
+                    f"PreProcessor couldn't find custom sentence tokenizer model for {self.language}. Using default {self.language} model."
+                )
+                return nltk.tokenize.sent_tokenize(text, language=language_name)
+
+            logger.error(
+                f"PreProcessor couldn't find default or custom sentence tokenizer model for {self.language}. Using English instead."
+            )
+            return nltk.tokenize.sent_tokenize(text, language="english")
+
+        # Use a default NLTK model
+        if language_name:
```

Comment:
And here too:
```suggestion
        if language_name is not None:
```

Reply:
Firstly, I have done this way, but @ZanSara have made this point:

> Just a minor thing here: language_name is not None will be True if language_name == "". With strings I think it's safer to check with not language_name, which will evaluate to False if language_name == "". I know that in the current code this can't happen, but I find it a bit more future-proof.

I have committed the `if x is not None` version again, please advise of any further changes,

Reply:
Sorry for bringing this up again, I think an explicit None check is just more clear about what we're checking here for. Without it it's kind of harder to understand what happens if someone passes `""` or even `[]` (especially for the model path). I would find it unintuitive for this method to just work as if the params hasn't been set in that cases.